### PR TITLE
addresses feedback on ap-2211, 2212, and 2106

### PR DESCRIPTION
**Pull request by noahamac**
https://looker.atlassian.net/browse/AP-2106
https://looker.atlassian.net/browse/AP-2212
https://looker.atlassian.net/browse/AP-2211

Would appreciate help testing this, too. Dev variant is ready to go on app.dev. Try to take this PR through the ringer... Whatever edge cases you can think of while testing, and whatever nits you have about the code. 

Thanks so much!

### CODE REVIEW

File: multiple_value.js [hunk 1]
```diff
@@ -57,4 +57,4 @@ and limitations under the License.
...
...hange"},{"Calculate Progress":"calculate_progress"},{"Calculate Progress (with Percentage)":"calculate_progress_perc"}],section:"Comparison",default:"value",order:10*t+1},h["comparison_show_label_".concat(e.name)]={type:"boolean",label:"".concat(e.label," - Show Label"),section:"Comparison",default:!0,order:10*t+2},n["comparison_show_label_".concat(e.name)]&&(h["comparison_label_".concat(e.name)]={type:"string",label:"".concat(e.label," - Label"),placeholder:e.label,section:"Comparison",order:10*t+3},h["comparison_label_placement_".concat(e.name)]={type:"string",label:"".concat(e.label," - Label Placement"),display:"select",values:[{Above:"above"},{Below:"below"},{Left:"left"},{Right:"right"}],default:"below",section:"Comparison",order:10*t+4})))})),l()(Co,h)&&l()(So,n)||(this.trigger("registerOptions",h),Co=Object.assign({},h),So=Object.assign({},n));var p={},d=-1,v=f.filter((function(e,t){return!0!==n["show_comparison_".concat(e.name)]?(d++,!0):(p[d]=t,!1)})).map((function(e,t){var n=p[t];return n&&(e.comparison=f[n]),e}));this.chart=a.a.render(i.a.createElement(_o,{config:n,data:v}),t),u()}else this.addError({title:"No Measures",message:"This chart requires measures"})}})}])}));
```
Comment (thomasbanghart) [inline multiple_value.js:60]: I think this dist file is missing the feature updates -- no biggie if we make sure the CDN version has it but might be a good idea to make sure what is here matches what we have on the CDN

File: src/multiple_value/ComparisonDataPoint.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  dataPoint: any,
+  percChange: number,
+  progressPerc: number,
+  handleClick: (i: any, j: any)=>{},
+}> = ({ config, compDataPoint, dataPoint, percChange, progressPerc, handleClick }) => {
+  return (
+    <ComparisonDataPointGroup>
+
+    {config[`comparison_style_${compDataPoint.name}`] !== 'percentage_change' ? null : (
+      <ComparisonPercentageChange value={percChange} onClick={() => { handleClick(compDataPoint, event) }}>
```
Comment (thomasbanghart) [inline src/multiple_value/ComparisonDataPoint.tsx:110]: resolve ts errors, no ts-ignore

File: src/multiple_value/ComparisonDataPoint.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        {percChange >= 0 ? <UpArrow pos={config[`pos_is_bad_${compDataPoint.name}`]}/> : <DownArrow pos={config[`pos_is_bad_${compDataPoint.name}`]}/>}
```
Comment (thomasbanghart) [inline src/multiple_value/ComparisonDataPoint.tsx:111]: resolve ts errors, no ts-ignore

File: src/multiple_value/ComparisonDataPoint.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    {config[`comparison_style_${compDataPoint.name}`] !== 'value' ? null : 
+    <ComparisonSimpleValue onClick={() => { handleClick(compDataPoint, event) }}>
+      {config[`comp_value_format_${compDataPoint.name}`] === "" ? compDataPoint.formattedValue : SSF.format(config[`comp_value_format_${compDataPoint.name}`], compDataPoint.value)}
+    </ComparisonSimpleValue>}
+
+    {config[`comparison_style_${compDataPoint.name}`] !== 'calculate_progress' &&
+    config[`comparison_style_${compDataPoint.name}`] !== 'calculate_progress_perc' ? null : (
+      <ComparisonProgressBar background={config[`style_${dataPoint.name}`]}>
+        <ComparisonProgressBarFilled
+          background={config[`style_${dataPoint.name}`]}
```
Comment (thomasbanghart) [inline src/multiple_value/ComparisonDataPoint.tsx:125]: resolve ts errors, no ts-ignore

File: src/multiple_value/multiple_value_container.js [hunk 5]
```diff
@@ -27,7 +27,7 @@ const baseOptions = {
     default: 'auto',
     order: 0,
     display_size: 'half'
-  }
+  },
```
Comment (thomasbanghart) [inline src/multiple_value/multiple_value_container.js:30]: 🙇

File: src/multiple_value/multiple_value_container.js [hunk 6]
```diff
@@ -96,6 +96,15 @@ looker.plugins.visualizations.add({
     
     dataPoints.forEach((dataPoint, index) => {
       //Style -- apply to all
+      if (config.orientation === "horizontal") {
```
Comment (thomasbanghart) [inline src/multiple_value/multiple_value_container.js:99]: Love the conditional options and the dividers look really nice 👍

**Review by thomasbanghart — CHANGES_REQUESTED** [review]
Correct me if I'm wrong, but at least when I tried to build this PR locally, those ts errors prevented the build so the dist file didn't update as expected. Maybe ts-ignore will suffice on those lines for now? 

Otherwise, everything looks great and addresses the FRs! 👍 Also resolves the issues we saw on db-next. Nice!

File: webpack.config.js [hunk 7]
```diff
@@ -9,23 +9,41 @@ var webpackConfig = {
   },
   devServer: {
     contentBase: './dist',
+    https: true,
+    headers: {
```
[the remark this answers is not in the document]
Comment (noahamac) [reply webpack.config.js:13]: Yes :pray:

File: multiple_value.js (hunk 1 above)
Comment (noahamac) [reply multiple_value.js:60]: Good catch

File: src/multiple_value/multiple_value_container.js (hunk 6 above)
Comment (noahamac) [reply src/multiple_value/multiple_value_container.js:99]: I actually changed the implementation away from borders and into self contained components... Let me know what u think of that

**Comment by noahamac** [comment]
Ok, cleaned some things up. Got rid of my ts-ignores :shudder:. You can find the viz on mp-viz: https://app.dev.looker.com/dashboards/710

Would appreciate another front-end test and code review @thomasbanghart

**Comment by thomasbanghart** [comment]
Awesome 👏  Thanks so much Noah!!

File: src/multiple_value/ComparisonDataPoint.tsx [hunk 8]
```diff
@@ -103,33 +106,43 @@ export const ComparisonDataPoint: React.FC<{
   progressPerc: number,
   handleClick: (i: any, j: any)=>{},
 }> = ({ config, compDataPoint, dataPoint, percChange, progressPerc, handleClick }) => {
+
+  function tryFormatting(formatString: string, value: number, defaultString: string) {
```
Comment (thomasbanghart) [inline src/multiple_value/ComparisonDataPoint.tsx:110]: Nice 👍

**Review by thomasbanghart — COMMENTED** [review]
LGTM!

**Comment by jcbmllgn** [comment]
Thanks for the continued work & maintenance of this package 🙌 🙇

**Pull request merged**